src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/calendar/PublicHolidaysCollectionTest.java

```diff
@@ -55,4 +62,34 @@ public void should_return_empty_list_with_null() {
         Set<LocalDate> oneCalendarResult = publicHolidaysCollection.getPublicHolidays(nullList);
         assertThat(oneCalendarResult.size()).isEqualTo(0);
     }
+
+    @Test
+    public void should_change_after_cache_expiry_external_api() {
+        BankHolidays resultFromApi = publicHolidaysCollection.getPublicHolidays(CALENDAR_URI);
+
+        PublicHolidaysCollectionTest.TestConfiguration.fakeTicker.advance(10, TimeUnit.HOURS);
+
+        BankHolidays resultFromCache = publicHolidaysCollection.getPublicHolidays(CALENDAR_URI);
+
+        assertThat(resultFromApi).isSameAs(resultFromCache);
+
+        PublicHolidaysCollectionTest.TestConfiguration.fakeTicker.advance(25, TimeUnit.HOURS);
+
+        BankHolidays resultFromRenewedCache = publicHolidaysCollection.getPublicHolidays(CALENDAR_URI);
+
+        assertThat(resultFromApi).isSameAs(resultFromCache).isNotSameAs(resultFromRenewedCache);
+    }
+
+    @Configuration
+    @Import(Application.class)
+    public static class TestConfiguration {
+
+        static FakeTicker fakeTicker = new FakeTicker();
+
+        @Bean
+        public Ticker ticker() {
+            return fakeTicker::read;
+        }
+
+    }
```

Comment:
can we add a test to cover AC4? I.e. when DMN has new calendar details, use this and not cache?